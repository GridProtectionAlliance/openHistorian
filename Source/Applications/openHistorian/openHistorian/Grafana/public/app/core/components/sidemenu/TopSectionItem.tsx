import React, { FC } from 'react';
import SideMenuDropDown from './SideMenuDropDown';
import { Icon } from '@grafana/ui';
import { NavModelItem } from '@grafana/data';

export interface Props {
  link: NavModelItem;
  onClick?: () => void;
}

const TopSectionItem: FC<Props> = props => {
  const { link, onClick } = props;

  return (
    <div className="sidemenu-item dropdown">
      <a className="sidemenu-link" href={link.url} target={link.target} onClick={onClick}>
        <span className="icon-circle sidemenu-icon">
          <Icon name={link.icon as any} size="xl" />
          {link.img && <img src={link.img} />}
        </span>
      </a>
      <SideMenuDropDown link={link} onHeaderClick={onClick} />
    </div>
  );
};

export default TopSectionItem;
