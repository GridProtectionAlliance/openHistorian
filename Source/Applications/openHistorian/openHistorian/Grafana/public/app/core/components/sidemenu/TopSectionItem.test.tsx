import React from 'react';
import { mount } from 'enzyme';
import TopSectionItem from './TopSectionItem';

const setup = (propOverrides?: object) => {
  const props = Object.assign(
    {
      link: {
        text: 'Hello',
        url: '/asd',
      },
    },
    propOverrides
  );

  return mount(<TopSectionItem {...props} />);
};

describe('Render', () => {
  it('should render component', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
