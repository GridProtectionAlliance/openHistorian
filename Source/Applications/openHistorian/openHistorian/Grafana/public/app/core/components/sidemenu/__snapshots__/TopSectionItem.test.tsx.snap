// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render should render component 1`] = `
<TopSectionItem
  link={
    Object {
      "text": "Hello",
      "url": "/asd",
    }
  }
>
  <div
    className="sidemenu-item dropdown"
  >
    <a
      className="sidemenu-link"
      href="/asd"
    >
      <span
        className="icon-circle sidemenu-icon"
      >
        <Icon
          size="xl"
        >
          <div />
        </Icon>
      </span>
    </a>
    <SideMenuDropDown
      link={
        Object {
          "text": "Hello",
          "url": "/asd",
        }
      }
    >
      <ul
        className="dropdown-menu dropdown-menu--sidemenu"
        role="menu"
      >
        <li
          className="side-menu-header"
        >
          <a
            className="side-menu-header-link"
            href="/asd"
          >
            <span
              className="sidemenu-item-text"
            >
              Hello
            </span>
          </a>
        </li>
      </ul>
    </SideMenuDropDown>
  </div>
</TopSectionItem>
`;
